include/seqan3/contrib/parallel/buffer_queue.hpp

```diff
@@ -191,46 +228,90 @@ class buffer_queue
     bool is_empty() const noexcept
     {
         std::unique_lock write_lock(mutex);
-        return headPos == tailPos;
+        return pop_front_position == push_back_position;
     }
 
     bool is_full() const noexcept
     {
         std::unique_lock write_lock(mutex);
-        return size_impl() == data.max_size();
+        return is_ring_buffer_exhausted(pop_front_position, push_back_position);
     }
 
     size_type size() const noexcept
     {
         std::unique_lock write_lock(mutex);
-        return size_impl();
+        size_type number_of_elements = push_back_position - pop_front_position;
```

Comment:
are you sure that this never turns negative and cause an underflow?
Maybe assert LEQ beforehand

Reply:
Boah, I need to think. But I guess you are right. In general we assume that pop_front position is always before the push_back position, but since both positions always increasing it could happen that suddenly the push_back_position overlfows starting from 0 again. this is ok from the algotithm but would here result in a wrong size. In general I do not see many useful things you can do with the size on the concurrent queue, but it should be correct anyway.